src/nbl/asset/utils/CMeshManipulator.cpp

```diff
@@ -130,21 +130,31 @@ void IMeshManipulator::flipSurfaces(ICPUMeshBuffer* inbuffer)
 
 core::smart_refctd_ptr<ICPUMeshBuffer> CMeshManipulator::createMeshBufferFetchOptimized(const ICPUMeshBuffer* _inbuffer)
 {
-	if (!_inbuffer || !_inbuffer->getPipeline() || !_inbuffer->getIndices())
-		return NULL;
+	if (!_inbuffer)
+		return nullptr;
 
-	auto outbuffer = createMeshBufferDuplicate(_inbuffer);
-    auto* outPipeline = outbuffer->getPipeline();
+    const auto* pipeline = _inbuffer->getPipeline();
+    const void* ind = _inbuffer->getIndices();
+	if (!pipeline || !ind)
+		return nullptr;
+
+	auto outbuffer = core::move_and_static_cast<ICPUMeshBuffer>(_inbuffer->clone(1u));
+    outbuffer->setAttachedDescriptorSet(core::smart_refctd_ptr<ICPUDescriptorSet>(const_cast<ICPUDescriptorSet*>(_inbuffer->getAttachedDescriptorSet())));
+    outbuffer->setSkin(
+        SBufferBinding<ICPUBuffer>(reinterpret_cast<const SBufferBinding<ICPUBuffer>&>(_inbuffer->getInverseBindPoseBufferBinding())),
```

Comment:
??? whats going on with the reinterpret_cast

Reply:
from what i see it already returns `const SBufferBinding<const BufferType>&`, why?

Reply:
it should be a `const_cast` but it cant cast away the const-ness of the `SBufferBinding<const BufferType>::buffer` pointer, so that's why